src/cli/scaffold/rust.rs

```diff
@@ -14,6 +23,61 @@ pub struct RustScaffold {
     build_template: Build,
 }
 
+/// Modify Cargo.toml in place, using pieces of the original
+fn rewrite_cargo_toml(base_path: &Path) -> DefaultResult<()> {
+    let cargo_file_path = base_path.join(CARGO_FILE_NAME);
+    let mut cargo_file = OpenOptions::new()
+        .read(true).write(true).open(cargo_file_path)?;
+    let mut contents = String::new();
+    cargo_file.read_to_string(&mut contents)?;
+
+    // create new Cargo.toml using pieces of the original
+    let zome_name = get_zome_name(base_path);
+    let new_toml = generate_cargo_toml(zome_name.as_str(), contents.as_str())?;
+    cargo_file.seek(SeekFrom::Start(0))?;
+    cargo_file.write_all(new_toml.as_bytes())?;
+    Ok(())
+}
+
+/// Assuming the base_path is at `[zomename]/code`, get the zome name
+fn get_zome_name(base_path: &Path) -> String {
+    base_path.parent()
+        .and_then(|p| p.to_str())
+        .unwrap_or("myzome")
+        .replace("/", "-")
+}
+
+/// Given existing Cargo.toml string, pull out some values and return a new
+/// string with values pulled from template
+fn generate_cargo_toml(name: &str, contents: &str) -> DefaultResult<String> {
+    let config: Value = toml::from_str(contents)?;
+
+    let authors_default = Value::from("[\"TODO\"]");
+    let edition_default = Value::from("\"TODO\"");
+    let maybe_package = config.get("package");
+
+    let name = Value::from(name);
+    let authors = maybe_package
+        .and_then(|p| p.get("authors"))
+        .unwrap_or(&authors_default);
+    let edition = maybe_package
```

Comment:
I haven't been seeing edition on the Cargo.toml files I've been looking at... maybe we don't want to reinforce this as a default?